### PR TITLE
fix: health check exception with opentelemetry tracing interceptors

### PR DESCRIPTION
**Pull request by girishc13**
The `async` tracing interceptors expect the RPC stubs to be a `coroutine` function otherwise an exception is thrown for the RPC. When tracing was enabled, the health checks with docker or k8s containers failed silently and the Flow timed out eventually. The underlying issue was the usage of the default `grpc_health.HealthServicer` implementation with an `async` grpc server.

Goals:
- use `aio.HealthServicer` and `await` keyword when calling the health servicer.
- add an instrumentation test with a docker container to address the fix.

### CODE REVIEW

File: jina/serve/instrumentation/_aio_server.py [hunk 1]
```diff
@@ -60,6 +60,8 @@ def telemetry_wrapper(behavior, request_streaming, response_streaming):
         return _wrap_rpc_behavior(next_handler, telemetry_wrapper)
 
     def _intercept_aio_server_unary(self, behavior, handler_call_details):
+        from jina.helper import iscoroutinefunction
```
Comment (JoanFM) [inline jina/serve/instrumentation/_aio_server.py:63]: why is this needed?

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/serve/instrumentation/_aio_server.py (hunk 1 above)
Comment (girishc13) [reply jina/serve/instrumentation/_aio_server.py:63]: Not needed if we decide to use the `aio.HealthServicer` approach. I was just experimenting if the interceptors can be easily modified to accept any method.

File: jina/serve/runtimes/asyncio.py [hunk 2]
```diff
@@ -182,6 +182,7 @@ def is_ready(ctrl_address: str, **kwargs) -> bool:
         :return: True if status is ready else False.
         """
 
+        # print('--->address', ctrl_address)
```
Comment (JoanFM) [inline jina/serve/runtimes/asyncio.py:185]: remember to remove

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Review by JoanFM — APPROVED** [review]
